### PR TITLE
SLING-12365 - Support for new servlet containers in Sling launchpad

### PR DESCRIPTION
**Pull request by sagarmiglani**
- This PR upgrade sling-launchpad base to use jakarta-servlet-api 5/6
- It also removes a few deprecated classes

### CODE REVIEW

**Comment by sagarmiglani** [comment]
@cziegeler Can you please review this PR?
Also, I added a few test cases, but coverage is still 0? 🤔

File: src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -17,10 +17,10 @@
   specific language governing permissions and limitations
   under the License.
 -->
-<web-app version="3.1"
-         xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+<web-app version="6.0"
```
Comment (cziegeler) [inline src/main/webapp/WEB-INF/web.xml:20]: This requires servlet API 6, maybe its better to use 5?

**Comment by cziegeler** [comment]
@sagarmiglani Code looks good to me - with the exception of the web.xml which seems to require servlet api 6
I guess there is something wrong with how the code coverage is measured in this project

File: src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (sagarmiglani) [reply src/main/webapp/WEB-INF/web.xml]: Updated it to use 5, thanks

**Review by cziegeler — APPROVED** [review]

File: src/test/java/org/apache/sling/launchpad/base/webapp/SlingServletDelegateTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        verify(servletResponse).sendError(HttpServletResponse.SC_NOT_FOUND);
+    }
+
+    @Test
+    public void testCalculateServletPackagesForVersionLessThan5() throws Exception {
+        String servletVersion = "3.1";
+        int majorVersion = 3;
+
+        // Access the private calculateServletPackages method
+        Method calculateServletPackagesMethod = SlingServletDelegate.class.getDeclaredMethod("calculateServletPackages", String.class, int.class);
```
Comment (rombert) [inline src/test/java/org/apache/sling/launchpad/base/webapp/SlingServletDelegateTest.java:64]: Nit: you could make the `SlingServerDelegate.calculateServletPackages` method package-protected and not have to rely on reflection.

AFAICT that package is not exported so there is no risk of leaking implementation details in the public API.

**Review by rombert — COMMENTED** [review]
Thanks for the contribution @sagarmiglani . 

My only question is - what happens with servlets implementing the javax.servlet APIs? All of Sling is currently using those and we switched directly to the jakarta.servlet ones.

**Comment by cziegeler** [comment]
@rombert The launchpad code runs inside a WAR file deployed into a servlet container supporting servlet API 5/6. The launchpad uses Apache Felix Http Bridge to support all OSGi specifications dealing with http/servlet inside the OSGi framework and that includes javax.servlet. So inside the framework all you need is that bridge bundle and a jar providing javax.servlet API and everything else continues to run as-is.
That's at least the theory :)

**Comment by rombert** [comment]
> @rombert The launchpad code runs inside a WAR file deployed into a servlet container supporting servlet API 5/6. The launchpad uses Apache Felix Http Bridge to support all OSGi specifications dealing with http/servlet inside the OSGi framework and that includes javax.servlet. So inside the framework all you need is that bridge bundle and a jar providing javax.servlet API and everything else continues to run as-is. That's at least the theory :)

Ah, that makese sense, thanks @cziegeler

File: src/test/java/org/apache/sling/launchpad/base/webapp/SlingServletDelegateTest.java (hunk 2 above)
Comment (sagarmiglani) [reply src/test/java/org/apache/sling/launchpad/base/webapp/SlingServletDelegateTest.java]: Thanks @rombert I have pushed the changes.

**Comment by sagarmiglani** [comment]
> My only question is - what happens with servlets implementing the javax.servlet APIs? All of Sling is currently using those and we switched directly to the jakarta.servlet ones.
@rombert As explained by @cziegeler, The Launchpad code runs within a WAR file deployed to a servlet container that supports Servlet API 5/6. Within this framework, we can utilize the Apache Felix HTTP Bridge bundle (5.1.x) and the Apache Felix HTTP Servlet API bundle (3.0.0) to ensure everything works smoothly. I ran some tests with these changes, and everything appears to be functioning well.

**Review by rombert — APPROVED** [review]

**Pull request merged**